internal/cluster/cluster.go

```diff
@@ -741,14 +746,22 @@ func (m *Manager) SetVipsData(ctx context.Context, c *common.Cluster, apiVip, in
 		db = m.db
 	}
 	log := logutil.FromContext(ctx, m.log)
+	formattedApiLease := network.FormatLease(apiVipLease)
+	formattedIngressVip := network.FormatLease(ingressVipLease)
+	if apiVip == c.APIVip &&
```

Comment:
Consider adding a comment saying there is nothing to update hence - return early.